python/triton/compiler.py

```diff
@@ -1725,19 +1735,23 @@ def compile(fn, **kwargs):
         for key in list(extern_libs):
             if extern_libs[key] == '' or extern_libs[key] is None:
                extern_libs.pop(key)
-        gfx_arch = os.environ.get('MI_GPU_ARCH', compile.discovered_gfx_arch)
+
+        gfx_arch_full_details = compile.discovered_gfx_arch_fulldetails
+        gfx_arch = os.environ.get('MI_GPU_ARCH', gfx_arch_full_details[1])
         if gfx_arch is None:
-            raise RuntimeError('gfx_arch is None (not specified)')
+            raise RuntimeError('gfx_arch is None (not specified)')        
         stages = {
             "ast": (lambda path: fn, None),
             "ttir": (lambda path: _triton.ir.parse_mlir_module(path, context),
                     lambda src: ast_to_ttir(src, signature, configs[0], constants)),
             "ttgir": (lambda path: _triton.ir.parse_mlir_module(path, context),
                     lambda src: ttir_to_ttgir(src, num_warps, num_stages, capability)),
             "llir": (lambda path: Path(path).read_text(),
-                    lambda src: ttgir_to_llir(src, extern_libs, capability)),
+                    lambda src: ttgir_to_llir(src, extern_libs, capability)),            
```

Comment:
Please remove the trialing spaces here.

Reply:
Done, and I have uniformed the prefix "gfx" for variable names.